app-modules/authorization/database/migrations/2024_12_30_142107_data_seed_role_permissions.php

```diff
@@ -0,0 +1,87 @@
+    see https://github.com/canyongbs/advisingapp/blob/main/LICENSE.
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use Database\Migrations\Concerns\CanModifyPermissions;
+use Illuminate\Database\Migrations\Migration;
+
+return new class () extends Migration {
+    use CanModifyPermissions;
+
+    private array $permissions = [
+        'role.*.delete' => 'Role',
+        'role.*.force-delete' => 'Role',
+        'role.*.restore' => 'Role',
+        'role.*.update' => 'Role',
+        'role.create' => 'Role',
+    ];
+
+    private array $permissionsToDelete = [
+        'role_group.view-any',
+        'role_group.create',
+        'role_group.*.view',
+        'role_group.*.update',
+        'role_group.*.delete',
+        'role_group.*.restore',
+        'role_group.*.force-delete',
+    ];
+
+    private array $guards = [
+        'web',
+        'api',
+    ];
+
+    public function up(): void
+    {
+        collect($this->guards)
+            ->each(function (string $guard) {
+                $permissions = Arr::except($this->permissions, keys: DB::table('permissions')
+                    ->where('guard_name', $guard)
+                    ->pluck('name')
+                    ->all());
+
+                $this->createPermissions($permissions, $guard);
+            });
+
+        collect($this->guards)
+            ->each(fn (string $guard) => $this->deletePermissions($this->permissionsToDelete, $guard));
+    }
+
+    public function down(): void
+    {
+        collect($this->guards)
+            ->each(fn (string $guard) => $this->deletePermissions(array_keys($this->permissions), $guard));
```

Comment:
I think this should reverse and create the permissions that were deleted, what do you think?

Reply:
I did not include this since these permissions have not been used for months, so having them missing would not cause issues. Additionally, the `up()` can be run multiple times and will not fail if permissions do not exist when they are deleted.

Please let me know if you still want me to recreate those permissions and I will do